tests/components/sensor/test_arlo.py

```diff
@@ -0,0 +1,185 @@
+"""The tests for the Netgear Arlo sensors."""
+import asyncio
+from collections import namedtuple
+import unittest
+from unittest.mock import patch, MagicMock
+from homeassistant.const import (
+    DEVICE_CLASS_TEMPERATURE, DEVICE_CLASS_HUMIDITY, ATTR_ATTRIBUTION)
+from homeassistant.components.sensor import arlo
+from homeassistant.components.arlo import DATA_ARLO
+from homeassistant.helpers import dispatcher
+from tests.common import get_test_home_assistant
+
+
+class TestArloSensor(unittest.TestCase):
+    """Test Netgear Arlo sensors."""
+
+    def _setup_arlo(self):
+        self.hass.data = {
+            DATA_ARLO: 'test'
+        }
+
+    @staticmethod
+    def _get_mock_sensor(name='Last', sensor_type='last_capture', data=None):
+        if data is None:
+            data = {}
+        return arlo.ArloSensor(name, data, sensor_type)
+
+    @staticmethod
+    def _get_named_tuple(input_dict):
+        return namedtuple('Struct', input_dict.keys())(*input_dict.values())
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """Setup shared dependencies for each test."""
+        self.hass = get_test_home_assistant()
+
+    def tearDown(self):  # pylint: disable=invalid-name
+        """Tear down shared dependencies for each test."""
+        self.hass.stop()
+
+    def test_setup_with_no_data(self):
+        """Test setup_platform with no data."""
+        result = arlo.setup_platform(self.hass, None, None)
+        self.assertEqual(result, False)
+
+    sensors = None
+
+    def _add_devices(self, sensors, boolean):
+        if boolean:
+            self.sensors = sensors
+
+    def test_setup_with_valid_data(self):
+        """Test setup_platform with valid data."""
+        config = {
+            'monitored_conditions': [
+                'last_capture',
+                'total_cameras',
+                'captured_today',
+                'battery_level',
+                'signal_strength',
+                'temperature',
+                'humidity',
+                'air_quality'
+            ]
+        }
+
+        self.hass.data[DATA_ARLO] = self._get_named_tuple({
+            'cameras': [self._get_named_tuple({
+                'name': 'Camera',
+                'model_id': 'ABC1000'
+            })],
+            'base_stations': [self._get_named_tuple({
+                'name': 'Base Station',
+                'model_id': 'ABC1000'
+            })]
+        })
+
+        arlo.setup_platform(self.hass, config, self._add_devices)
+        self.assertEqual(len(self.sensors), 8)
+
+    def test_sensor_name(self):
+        """Test the name property."""
+        sensor = TestArloSensor._get_mock_sensor()
+        self.assertEqual(sensor.name, 'Last')
+
+    @asyncio.coroutine
```

Comment:
Remove this.

Reply:
Again, having just come back to python (after several years and from python2), async methods in python3 have come quite a long way, and I'm still getting accustomed to the new stuff:

When I omit this, it complains that the function is never awaited. In tests for other HA components, I noticed that standalone functions omit this, but functions declared in classes have it. Is this correlation true or just coincidence? If it does in fact need to be removed, what's the appropriate way to address the no-await issue?

Reply:
We're already declaring the method a coroutine by saying `async def`. This is the new async syntax in Python 3.5. So we shouldn't add the coroutine decorator on top of that. Class based unittest tests don't support coroutines as test methods. Here we have to move to standalone pytest tests.